src/azure-cli/azure/cli/command_modules/profile/custom.py

```diff
@@ -136,6 +151,9 @@ def login(cmd, username=None, password=None, service_principal=None, tenant=None
         from azure.cli.core.auth.identity import ServicePrincipalAuth
         password = ServicePrincipalAuth.build_credential(password, client_assertion, use_cert_sn_issuer)
 
+    select_subscription = interactive and sys.stdin.isatty() and sys.stdout.isatty() and \
+        cmd.cli_ctx.config.getboolean('core', 'login_experience_v2', fallback=True)
```

Comment:
This is long logic, how about creating a function.

If `service_principal` is not None, the `select_subscription` is also enabled? This is conflict with PR description.

Reply:
True, but not that complex as it is still a one-liner.

Reply:
> This is long logic, how about creating a function?

True, but not that complex as it is still only one line.

> If `service_principal` is not None, the `select_subscription` is also enabled? This conflicts with PR description.

`service_principal` must be used together with `username`:

https://github.com/Azure/azure-cli/blob/3b20c9ad3ac93da8a707198f530397c09412279a/src/azure-cli/azure/cli/command_modules/profile/custom.py#L127-L128

`interactive` is `True` only when `username` is not provided:

https://github.com/Azure/azure-cli/blob/3b20c9ad3ac93da8a707198f530397c09412279a/src/azure-cli/azure/cli/command_modules/profile/custom.py#L141-L148

In other words, when `username` is provided, `interactive` must be `False`.

Reply:
How about `az login --identity`?

Reply:
It is guarded by 

https://github.com/Azure/azure-cli/blob/4f151800433c8d2cdfc4dbcab4e99da7217a203e/src/azure-cli/azure/cli/command_modules/profile/custom.py#L131-L136